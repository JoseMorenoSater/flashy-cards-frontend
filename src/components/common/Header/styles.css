.nav {
  background-color: #00a6ff;
}

.logo {
  color: white;
}
