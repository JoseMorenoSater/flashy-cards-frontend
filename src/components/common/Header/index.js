/**
 *
 * Header
 *
 */
import React from "react";
import { Link } from "react-router-dom";
import "./styles.css";

const Header = ({ isAuthenticated, username, onLogoutClick }) => {
  return (
    <nav className="navbar nav">
      <Link
        to="/dashboard"
        className="navbar-brand logo                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   "
      >
        Flashy Cards
      </Link>
      {isAuthenticated && (
        <div className="user">
          {username} |{" "}
          {
            <span className="logout" onClick={onLogoutClick}>
              logout
            </span>
          }
        </div>
      )}
    </nav>
  );
};

export default Header;
