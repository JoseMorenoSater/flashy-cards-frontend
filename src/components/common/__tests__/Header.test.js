import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Header from "../Header";

describe("Header Component", () => {
  it("should render header correctly", () => {
    const HeaderComponent = create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    ).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });
});
