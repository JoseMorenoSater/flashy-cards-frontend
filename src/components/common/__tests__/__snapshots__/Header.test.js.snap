// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render header correctly 1`] = `
<nav
  className="navbar nav"
>
  <a
    className="navbar-brand logo                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   "
    href="/dashboard"
    onClick={[Function]}
  >
    Flashy Cards
  </a>
</nav>
`;
